RESEARCH.md

```diff
@@ -0,0 +1,95 @@
+
+NFC technology research
+===
+
+In order to make sure that the Web NFC API is future-proof, it is important to have a good overview of all the technologies that as part of NFC as well as those that support it. There are many technologies that are used by NFC but has broader usage and there are technologies not defined in the NFC Forum that people consider part of NFC.
+
+This document serves as a live document for our current understanding on the NFC ecosystem.
+
+
+NFC Simple Exchange Protocol
+---
+The SNEP is a simple protocol for doing peer-to-peer (P2P) NDEF transfers. It is not supported by CoreNFC (iOS) and it is being deprecated on Android (where is branded Android Beam) due to security issues).
+
+The only way to allow some kind of peer-to-peer, like pushing NDEF from active device to active device, or even establishing a two channel communication channel, is by using HCE.
+
+### Conclusion
+
+Due to SNEP being deprecated on Android and not supported by CoreNFC, Web NFC will not be adding P2P supporting over SNEP.
+
+Host Card Emulation
+---
+The idea of HCE was originally introduced on BlackBerry devices but today it is supported by Android out of the box.
+
+On Android, a NFC T4T (type 5 tag, no optional NFC-B support) is emulated, meaning ISO14433-1 as physical layer, ISO14443-2 as RF signal interface, ISO14443-3A (NFC-A) as activation and anti-collision, and ISO14443-4 (ISO-DEP) as transmission protocol, and ISO7816-4 (APDU) as card organization and structure (NDEF can be emulated on top of this).
+
+NFC Value Added Services
+---
+iOS supports NFC Value Added Services (VAS) [**Pending research**]
+
+MiFare support
+---
+NXP MiFare cards are quite popular and exists in different generations, with some NFC support, but also with support for proprietary protocols. As many phones have NFC readers from NXP, these additionally supports MiFare proprietary protocols. As iPhone’s NFC reader is also from NXP, CoreNFC supports MiFare protocols in addition to NFC protocols.
+
+MiFare Classic and Plus cards can be configured to support NDEF, but are not NFC compatible, meaning they do require a NFC reader chip from NXP.
+
+CoreNFC doesn’t support MiFare Classic (proprietary protocols) but people have had success reading NDEF formatted MiFare Classic cards.
+
```

Comment:
Link to what is CoreNFC?
Maybe some more explanation would not harm, i.e. how MIFARE exposes NDEF, but it's up to you.